internal/sequencerapi/api.go

```diff
@@ -0,0 +1,106 @@
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	sendRawTxConditionalCostMeter       = metrics.NewRegisteredMeter("sequencer/sendRawTransactionConditional/cost", nil)
+	sendRawTxConditionalRequestsCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/requests", nil)
+	sendRawTxConditionalAcceptedCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/accepted", nil)
+)
+
+type sendRawTxCond struct {
+	b      ethapi.Backend
+	seqRPC *rpc.Client
+}
+
+func GetSendRawTxConditionalAPI(b ethapi.Backend, seqRPC *rpc.Client) rpc.API {
+	return rpc.API{
+		Namespace: "eth",
+		Service:   &sendRawTxCond{b, seqRPC},
+	}
+}
+
+func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond types.TransactionConditional) (common.Hash, error) {
+	sendRawTxConditionalRequestsCounter.Inc(1)
+
+	cost := cond.Cost()
+	sendRawTxConditionalCostMeter.Mark(int64(cost))
+	if cost > params.TransactionConditionalMaxCost {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("conditional cost, %d, exceeded max: %d", cost, params.TransactionConditionalMaxCost),
+			Code:    params.TransactionConditionalCostExceededMaxErrCode,
+		}
+	}
+
+	// Perform sanity validation prior to state lookups
+	if err := cond.Validate(); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed conditional validation: %s", err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	state, header, err := s.b.StateAndHeaderByNumber(context.Background(), rpc.LatestBlockNumber)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	if err := header.CheckTransactionConditional(&cond); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed header check: %s", err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+	if err := state.CheckTransactionConditional(&cond); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed state check: %s", err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	// State is checked against an older block to remove the MEV incentive for this endpoint compared with sendRawTransaction
+	parentBlock := rpc.BlockNumberOrHash{BlockHash: &header.ParentHash}
+	parentState, _, err := s.b.StateAndHeaderByNumberOrHash(context.Background(), parentBlock)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	if err := parentState.CheckTransactionConditional(&cond); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed parent block %s state check: %s", header.ParentHash, err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	// forward if seqRPC is set, otherwise submit the tx
+	if s.seqRPC != nil {
+		var hash common.Hash
```

Comment:
One thing that i am noticing is that we don't do the fee check in this branch

https://github.com/ethereum-optimism/op-geth/blob/60038121c7571a59875ff9ed7679c48c9f73405d/internal/ethapi/api.go#L1997

Reply:
oh good call, I can add these preconditions here as well. Not ideal to dup that code but an alternative approach would be a bigger diff